catkin_tools_document/builders.py

```diff
@@ -83,7 +83,45 @@ def sphinx(conf, package, deps, output_path, source_path, docs_build_path):
     ]
 
 
+def pydoctor(conf, package, deps, output_path, source_path, docs_build_path):
+    output_dir = os.path.join(output_path, 'html', conf.get('output_dir', ''))
+
+    # TODO: Would be better to extract this information from the setup.py, but easier
+    # for now to just codify an assumption of {pkg}/python, falling back to {pkg}/src.
+    src_dir = os.path.join(source_path, 'python')
+    if not os.path.exists(src_dir):
+        src_dir = os.path.join(source_path, 'src')
+
+    command = [which('pydoctor'), '--project-name', package.name, '--html-output', output_dir]
+
+    for subdir in os.listdir(src_dir):
+        command.extend(['--add-package', package.name])
+
+    if 'config' in conf and 'epydoc' not in conf['config']:
+        command.extend(['--config', os.path.join(source_path, conf['config'])])
+
+    # pydoctor returns error codes for minor issues we don't care about.
+    wrapper_command = ['/bin/bash', '-c', '%s || true' % ' '.join(command)]
+
+    return [
+        FunctionStage(
+            'mkdir_pydoctor',
+            makedirs,
+            path=output_dir),
+        CommandStage(
+            'rosdoc_pydoctor',
+            wrapper_command,
+            cwd=src_dir)
+    ]
+
+
 def epydoc(conf, package, deps, output_path, source_path, docs_build_path):
+    try:
+        which('epydoc')
```

Comment:
Not sure if this is really the right approach here; another way to go would be just checking `sys.executable`, but that isn't really correct either. Maybe it should just be a `--force-pydoctor` flag or something?

Reply:
This looks reasonable to me.

Reply:
The main issue is it will fail out if you build/document a `PYTHON_EXECUTABLE=python3` workspace on a machine where `python2-epydoc` is installed. Either epydoc will have to be uninstalled, or the packages in question updated to explicitly use `pydoctor`.

Reply:
But this is fallback logic anyway, so I think it's probably fine.